### ISSUE TITLE
Sum of two columns and group by

### ISSUE THREAD

**Issue by Pedramrn** [Raizlabs/DBFlow#346]
Hi, 

I want to get the sum of two columns and group them by another column. In SQL this is like:

```sql
SELECT groupColumn,
     SUM(upVote) as totalUpvote,
     SUM(downVote) as totalDownVote,
     (SUM(upVote)  -  SUM(downVote)) as totalVote
     FROM someTable
     GROUP BY groupColumn
     ORDER BY totalVote DESC;
```
How can I do this by DBFlow?
Dose DBFlow support raw SQL query?

Great job by the way.

**Comment by agrosner** [comment]
thanks @Pedramrn I apologize for the late response (multiple reasons). A query like this will be fully supported in `3.0.0` of this library in a very clean way.

as for now you could write something like:
```
new Select(MyTable$Table.GROUP_COLUMN, 
  "SUM(upvote) as totalUpvote",
  "SUM(downVote) as totalDownVote", 
 "(SUM(upVote)  -  SUM(downVote)) as totalVote")
.from(MyTable.class).groupBy("groupColumn").orderBy("totalVote DESC").queryCustomList(CustomQueryClass.class);
``` 

For this query, you'll need to create a `@QueryModel` class.

**Comment by Pedramrn** [comment]
@agrosner  Oh It's OK.
Can't wait for Version 3, sounds awesome. 

I tried your suggested solution but I couldn't figured out what is the `MyTable$Table.GROUP_COLUMN` so I put the the ID field for the moment. however I got no such a column exception.
 
This is my Model class:
```
@Table(databaseName = MyDatabase.NAME)
public class DailyUsageApps extends BaseModel {

    @Column
    @PrimaryKey(autoincrement = true)
    private long id;

    @Column
    private Date date;

    @Column
    private int uid;

    @Column
    private long wifiSent;

    @Column
    private long wifiReceived;

    @Column
    private long mobileSent;

    @Column
    private long mobileReceived;
}
```

QueryModel Class:

```
@QueryModel(databaseName = InternetMeterDatabase.NAME)
public class AppsQueryModel extends BaseQueryModel {

    @Column
    int uid;
    @Column
    long totalWifiReceived;
    @Column
    long totalWifiSent;
    @Column
    long totalMobileReceived;
    @Column
    long totalMobileSent;
    @Column
    long totalUsage;
}
```
I skipped getters and setters...

This is the query:
```
List<AppsQueryModel> queryModelList = new Select(DailyUsageApps$Table.ID,
                        "SUM(wifiReceived) as totalWifiReceived",
                        "SUM(wifiSent) as totalWifiSent",
                        "(SUM(wifiSent)  +  SUM(wifiReceived)) as totalUsage")
                        .from(DailyUsageApps.class).where().groupBy("uid").orderBy("totalUsage DESC").queryCustomList(AppsQueryModel.class);
```

And this the exeption which says no such a column:
```
08-21 19:55:20.060  30923-30923/com.mrn.internetmeter.ALPHA.debug E/SQLiteLog﹕ (1) no such column: SUM(wifiReceived) as totalWifiReceived
08-21 19:55:20.062  30923-30923/com.mrn.internetmeter.ALPHA.debug D/AndroidRuntime﹕ Shutting down VM
08-21 19:55:20.285  30923-30923/com.mrn.internetmeter.ALPHA.debug E/AndroidRuntime﹕ FATAL EXCEPTION: main
    Process: com.mrn.internetmeter.ALPHA.debug, PID: 30923
    android.database.sqlite.SQLiteException: no such column: SUM(wifiReceived) as totalWifiReceived (code 1): , while compiling: SELECT `id`, `SUM(wifiReceived) as totalWifiReceived`, `SUM(wifiSent) as totalWifiSent`, `(SUM(wifiSent)  +  SUM(wifiReceived)) as totalUsage` FROM `DailyUsageApps` GROUP BY uid  ORDER BY totalUsage DESC
            at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
            at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:889)
            at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:500)
            at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:588)
            at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:58)
            at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
            at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:44)
            at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1316)
            at android.database.sqlite.SQLiteDatabase.rawQuery(SQLiteDatabase.java:1255)
            at com.raizlabs.android.dbflow.sql.SqlUtils.queryList(SqlUtils.java:51)
            at com.raizlabs.android.dbflow.sql.language.BaseModelQueriable.queryCustomList(BaseModelQueriable.java:65)
            at com.mrn.internetmeter.views.fragments.FragmentDailyUsage_Sqlite.onCreateView(FragmentUsage.java:101)
            at android.support.v4.app.Fragment.performCreateView(Fragment.java:1789)
            at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:955)
            at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1138)
            at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:740)
            at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1501)
            at android.support.v4.app.FragmentManagerImpl.executePendingTransactions(FragmentManager.java:490)
            at android.support.v4.app.FragmentPagerAdapter.finishUpdate(FragmentPagerAdapter.java:141)
            at android.support.v4.view.ViewPager.populate(ViewPager.java:1105)
            at android.support.v4.view.ViewPager.populate(ViewPager.java:951)
            at android.support.v4.view.ViewPager$3.run(ViewPager.java:250)
            at android.view.Choreographer$CallbackRecord.run(Choreographer.java:767)
            at android.view.Choreographer.doCallbacks(Choreographer.java:580)
            at android.view.Choreographer.doFrame(Choreographer.java:549)
            at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:753)
            at android.os.Handler.handleCallback(Handler.java:739)
            at android.os.Handler.dispatchMessage(Handler.java:95)
            at android.os.Looper.loop(Looper.java:135)
            at android.app.ActivityThread.main(ActivityThread.java:5254)
            at java.lang.reflect.Method.invoke(Native Method)
            at java.lang.reflect.Method.invoke(Method.java:372)
            at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
            at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

So what am I doing wrong?

Thank you

**Comment by agrosner** [comment]
fixed in 3.0.0. as for that one, it was adding ` marks to the query.
